File: fixtures/csharp/type_inference/visibility/Visibility.cs
namespace MyApp.Models
{
    public class Visibility
    {
        public string PubField = "public";
        internal string InternalField = "internal";
        protected string ProtectedField = "protected";
        private string PrivateField = "private";

        public string GetPublic()
        {
            return PubField;
        }

        internal string GetInternal()
        {
            return InternalField;
        }

        protected string GetProtected()
        {
            return ProtectedField;
        }

        private string GetPrivate()
        {
            return PrivateField;
        }

        public string Describe()
        {
            return GetPublic() + GetInternal() + GetProtected() + GetPrivate();
        }
    }
}

File: fixtures/typescript/type_inference/visibility/visibility.ts
export class Visibility {
    public pubField: string = "public";
    protected protectedField: string = "protected";
    private privateField: string = "private";
    readonly readonlyField: string = "readonly";

    public getPublic(): string {
        return this.pubField;
    }

    protected getProtected(): string {
        return this.protectedField;
    }

    private getPrivate(): string {
        return this.privateField;
    }

    public describe(): string {
        return this.getPublic() + this.getProtected() + this.getPrivate();
    }
}
